### PR TITLE
Ports "Smooth movement - "TG coders suck" edition" from yogstation 13 (again)

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports [Yogstation#8132](https://github.com/yogstation13/Yogstation/pull/8132)
Brings over the buttery movement, and makes mobs with slow movement speeds glide instead of jump from tile to tile
Small modifications taken from [tgstation#52515](https://github.com/tgstation/tgstation/pull/52515)
Moving at high speeds no longer looks like you're teleporting

## Why It's Good For The Game
Movement feels a lot smoother and moving things wont be as stupid hard to click

## Changelog
:cl: monster860, actioninja, lordScrubling
add: Smooth movement from yogstation
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
someone already tried to port this I think? #1513

**Comment by BrumalLake** [comment]
what went wrong
oh, a vote
maybe people have changed since then

**Comment by BrumalLake** [comment]
This is immediately going to get closed because of the vote isnt it
dont know why people hate this

**Comment by utilbird** [comment]
@lordScrubling Because it makes the game uncomfortable to play at higher FPS

**Comment by Sinestia** [comment]
> 
> 
> @lordScrubling Because it makes the game uncomfortable to play at higher FPS

How? I'd wager it'd make it better for ultra high FPS setups. 

This is probably just boomer propaganda

**Comment by eLeCtrOssSnake** [comment]
Smooth movement is one of the reasons I don’t like playing on TG. It feels childish.

**Comment by eLeCtrOssSnake** [comment]
> > @lordScrubling Because it makes the game uncomfortable to play at higher FPS
> 
> How? I'd wager it'd make it better for ultra high FPS setups.
> 
> This is probably just boomer propaganda

No it actually slows it down in a weird looking way.

**Comment by SinfulBliss** [comment]
It's weird at first but it's way better and more logical after you warm up to it.

**Comment by eLeCtrOssSnake** [comment]
No, I still can’t bear it. I tried.

**Comment by utilbird** [comment]
> How? I'd wager it'd make it better for ultra high FPS setups.
> This is probably just boomer propaganda

#1513
Even monster themselves said it 😛

File: code/__DEFINES/movement.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// Originally a really stupid /tg/ var that sucked and was really bad and caused it to look horrible. Now it's a way of compensating for time dilation
+GLOBAL_VAR_INIT(glide_size_multiplier, 1.0)
+
+///Broken down, here's what this does:
+/// divides the world icon_size (32) by delay divided by ticklag to get the number of pixels something should be moving each tick.
+/// The division result is given a min value of 1 to prevent obscenely slow glide sizes from being set
+/// Then that's multiplied by the global glide size multiplier. 1.25 by default feels pretty close to spot on. This is just to try to get byond to behave.
+/// The whole result is then clamped to within the range above.
+/// Not very readable but it works
+#define DELAY_TO_GLIDE_SIZE(delay) (CLAMP(((32 / max((delay) / world.tick_lag, 1)) * GLOB.glide_size_multiplier), MIN_GLIDE_SIZE, MAX_GLIDE_SIZE))
```
Comment (utilbird) [inline code/__DEFINES/movement.dm:16]: This is quite expensive

**Comment by utilbird** [comment]
I believe tgstation/tgstation#52515 fixed some of these issues though

File: code/__DEFINES/movement.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/__DEFINES/movement.dm:16]: its literally just multiplication, although it could probably be cached when the move delay changes

**Comment by BrumalLake** [comment]
52515 didnt change anything much, just made the diagonal movement checks simpler to understand, removed the sloth's previous glide and changed something about orbiting

**Comment by BrumalLake** [comment]
bruh some dude put a single space after some if statements and created a conflict

**Comment by ike709** [comment]
We'll try it for a few rounds.

**Comment by francinum** [comment]
This feels like shit at 40 and gets sickening the higher you go.

**Comment by ike709** [comment]
I personally like it but it does make me feel physically ill yeah.

**Comment by BrumalLake** [comment]
doesn't seem to work very well with 50% td which is average it seems

**Pull request closed without merging**